pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -430,6 +430,20 @@ func validateInstancesDistribution(ng *NodeGroup) error {
 		return fmt.Errorf("spotInstancePools should be between 1 and 20")
 	}
 
+	if distribution.SpotInstancePools != nil && distribution.SpotAllocationStrategy != nil {
+		return fmt.Errorf("spotInstancePools cannot be specified when also specifying spotAllocationStrategy")
```

Comment:
Not entirely sure the check correlates with the message, or what's the intention of the error message: "spotInstancePools cannot be specified when also specifying spotAllocationStrategy"

`spotInstancePools` cannot be used when the `spotAllocationStrategy` is set to `capacity-optimized`. This has been captured right in the documentation (change in this PR over 08-spot-instances.md )

spotInstancePools is only intended to be used with lowes-price allocation strategy.  That, in my opinion should be a check to do at this stage, but: Is that what the conditional or the error message reflect ?

Reply:
I'll update with a more specific check and modify the error message returned.